pkg/admissioncontroller/webhooks/admission/seedrestriction/admission.go

```diff
@@ -202,6 +207,40 @@ func (h *handler) admitLease(seedName string, request admission.Request) admissi
 	return h.admit(seedName, &request.Name)
 }
 
+func (h *handler) admitSecret(ctx context.Context, seedName string, request admission.Request) admission.Response {
+	if request.Operation != admissionv1.Create {
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("unexpected operation: %q", request.Operation))
+	}
+
+	// Check if the secret is related to a BackupBucket assigned to the seed the gardenlet is responsible for.
+	if strings.HasPrefix(request.Name, v1beta1constants.SecretPrefixGeneratedBackupBucket) {
+		backupBucket := &gardencorev1beta1.BackupBucket{}
+		if err := h.cacheReader.Get(ctx, kutil.Key(strings.TrimPrefix(request.Name, v1beta1constants.SecretPrefixGeneratedBackupBucket)), backupBucket); err != nil {
+			if apierrors.IsNotFound(err) {
+				return admission.Errored(http.StatusForbidden, err)
+			}
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+
+		return h.admit(seedName, backupBucket.Spec.SeedName)
+	}
+
+	// Check if the secret is related to a Shoot assigned to the seed the gardenlet is responsible for.
+	if shootName, ok := gutil.IsShootProjectSecret(request.Name); ok {
```

Comment:
Can you elaborate why we need to take care about this in the SeedRestriction as well when we already add it to the graph [here](https://github.com/gardener/gardener/pull/4044/commits/65e9fcfa5bb3a183089ee775a4baa12ee0414bb3#diff-2e67e10c16922f29f77c38ccf2e57292d0f9b5914fb568dcf0c33569d041e2a3R139)?

Reply:
The graph is not consulted for `create` operations due to https://github.com/gardener/gardener/pull/4044/commits/a4204ccdc23ee88aa3a5ec85c1568942795294ac#diff-57b48b83bc91794df3bfa1273a97f83ec122063edb925d43c33f380ecb4b2b2eR246-R250. Yes, I could make it this way, but I didn't want to introduce more complexity/special handling for `Secret`s.

Reply:
Sorry, correction: Actually, I cannot make it this way because for `create` operations the object name is not known.

Reply:
Okay, thanks for reminding.